core/trino-main/src/main/java/io/trino/connector/system/MaterializedViewsSystemTableAdapter.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.connector.system;
+
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.predicate.TupleDomain;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Adapter to convert a materialized view representation into a single
+ * row to be displayed on the system metadata table.
+ */
+public interface MaterializedViewsSystemTableAdapter
```

Comment:
MaterializedViewsSystemTableAdapter has already been used in the previous commit.
We need every commit to compile cleanly and not break tests.
This commit doesn't make much sense on it's own, you can make it part of the previous commit.